### ISSUE TITLE
transaction processing issues

### ISSUE THREAD

**Issue by grd2345** [balderdashy/sails-postgresql#169]
I am trying to test transaction processing in postgresql and am having issues making this work.
I created a simple promise below to test rollback and it doesn't rollback.
I get WARNING:  there is no transaction in progress in the logs.
what is the correct way to do transaction processing? This is not working.

```
Facility.query('BEGIN;',function() {
     facilityService.CreateFacility(params)
      .then(function(facility) {
          Facility.query('ROLLBACK;',function() {
            ret.resolve(facility);
          })
      }) 
})
```

**Comment by devinivy** [comment]
See here: http://stackoverflow.com/questions/25079408/how-to-handle-async-concurrent-requests-correctly/25100188#25100188

It may be possible due to connection pooling that you're beginning and rolling back on different connections.  I'm not sure the best way to configure that!  Worth checking-out techniques used by https://www.npmjs.com/package/sails-mysql-transactions .

**Comment by kevinburkeshyp** [comment]
yep - basically no way to do this with waterline, we had to use the native `pg` library for this. hoping to write more about it soon.

**Comment by particlebanana** [comment]
We just need to expose a connection like the `.native` command in the mongo adapter. So you have `.query()` to run a sql query and another function that returns a connection object.

**Comment by fundon** [comment]
@particlebanana Do you have an example?

**Comment by sailsbot** [comment]
Thanks for posting, @grd2345.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by kevinburkeshyp** [comment]
Here's the library we use - https://github.com/Shyp/pg-transactions
